tests/test_agents.py

```diff
@@ -18,35 +18,31 @@
 
 import pytest
 
+from cloudai._core.configurator.cloudai_gym import CloudAIGymEnv
 from cloudai._core.configurator.grid_search import GridSearchAgent
-from cloudai._core.test_scenario import TestRun
 
 
 @pytest.fixture
-def mock_test_run():
+def mock_env():
     """
-    Fixture to provide a mock TestRun object for testing.
+    Fixture to provide a mock CloudAIGymEnv object for testing.
     """
-    test_run = MagicMock(spec=TestRun)
-    test_run.test = MagicMock()
-    test_run.test.cmd_args = {
-        "docker_image_url": "https://docker/url",
+    env = MagicMock(spec=CloudAIGymEnv)
+    env.define_action_space.return_value = {
         "iters": [10, 100],
```

Comment:
BTW, the reason you have to heavily mock is exactly because of confusing interface. If `define_action_space()` wouldn't be called in constructor, you could create the object and then do `env.action_space = {...}`.

Please consider reducing number of permutations, there is no value in more then 6 combinations, but the more we have the harder it is to read.